gui/journal.lua

```diff
@@ -0,0 +1,797 @@
+        x=#self.lines[#self.lines],
+        y=#self.lines
+    }
+end
+
+function TextEditorView:onRenderBody(dc)
+    dc:pen({fg=self.text_pen, bg=COLOR_RESET, bold=true})
+
+    local max_width = dc.width
+    local new_line = self.debug and NEWLINE or ''
+
+    for ind, line in ipairs(self.lines) do
+        -- do not render new lines symbol
+        local line = line:gsub(NEWLINE, new_line)
+        dc:string(line)
+        dc:newline()
+    end
+
+    local show_focus = not self:hasSelection()
+        and self.parent_view.focus
+        and gui.blink_visible(530)
+    if (show_focus) then
+        dc:seek(self.cursor.x - 1, self.cursor.y - 1)
+            :char('_')
+    end
+
+    if self:hasSelection() then
+        local sel_new_line = self.debug and PERIOD or ''
+        local from, to = self.cursor, self.sel_end
+        if (from.y > to.y or (from.y == to.y and from.x > to.x)) then
+            from, to = to, from
+        end
+
+        local line = self.lines[from.y]
+            :sub(from.x, to.y == from.y and to.x or nil)
+            :gsub(NEWLINE, sel_new_line)
+        dc:pen({ fg=self.text_pen, bg=self.select_pen })
+            :seek(from.x - 1, from.y - 1)
+            :string(line)
+
+        for y = from.y + 1, to.y - 1 do
+            line = self.lines[y]:gsub(NEWLINE, sel_new_line)
+            dc:seek(0, y - 1)
+                :string(line)
+        end
+
+        if (to.y > from.y) then
+            local line = self.lines[to.y]
+                :sub(1, to.x)
+                :gsub(NEWLINE, sel_new_line)
+            dc:seek(0, to.y - 1)
+                :string(line)
+        end
+
+        dc:pen({fg=self.text_pen, bg=COLOR_RESET})
+    end
+
+    if self.debug then
+        local cursor_char = self:charAtCursor()
+        local debug_msg = string.format(
+            'x: %s y: %s ind: %s #line: %s char: %s',
+            self.cursor.x,
+            self.cursor.y,
+            self:cursorToIndex(self.cursor.x, self.cursor.y),
+            #self.lines[self.cursor.y],
+            (cursor_char == NEWLINE and 'NEWLINE') or
+            (cursor_char == ' ' and 'SPACE') or
+            (cursor_char == '' and 'nil') or
+            cursor_char
+        )
+        local sel_debug_msg = self.sel_end and string.format(
+            'sel_end_x: %s sel_end_y: %s',
+            self.sel_end.x,
+            self.sel_end.y
+        ) or ''
+        dc:pen({fg=COLOR_LIGHTRED, bg=COLOR_RESET})
+            :seek(0, self.parent_view.frame_body.height - 1)
+            :string(debug_msg)
+            :seek(0, self.parent_view.frame_body.height - 2)
+            :string(sel_debug_msg)
+    end
+end
+
+function TextEditorView:charAtCursor()
+    local cursor_ind = self:cursorToIndex(self.cursor.x, self.cursor.y)
+    return self.text:sub(cursor_ind, cursor_ind)
+end
+
+function TextEditorView:getMultiLeftClick()
+    local from_last_click_ms = (os.clock() - (self.last_click or 0)) * 1000
+
+    if (from_last_click_ms > 250) then
```

Comment:
Please use the system preference for multi click ms. example: https://github.com/DFHack/dfhack/blob/develop/library/lua/gui/widgets.lua#L2643

Reply:
You mean `DOUBLE_CLICK_MS`? Its hardcoded to 500 in your example.

Reply:
500 is just the default value. The player-configured value is restored by `control-panel` when DF starts up

Reply:
Interesting. Its hard to spot from anaylizing only widgets module, but now I see it.